src/main/java/com/extrahardmode/features/AnimalCrowdControl.java

```diff
@@ -0,0 +1,173 @@
+                        new BukkitRunnable() {
+
+                            boolean firstRun = true;
+
+                            @Override
+                            public void run() {
+                                List<Entity> cattle = e.getNearbyEntities(3, 3, 3);
+                                int density = 0;
+
+                                //this will be used to check if animal is far from other animals
+                                for (Entity a : cattle) {
+                                    if (a instanceof Animals
+                                            && a.getType() != EntityType.HORSE
+                                            && a.getType() != EntityType.WOLF
+                                            && a.getType() != EntityType.OCELOT) {
+                                        density++;
+                                    }
+                                }
+                                
+                                if (animal.isDead() || density <= threshold) {
+                                    this.cancel();
+                                } else {
+                                    /**
+                                     *Hack to force animal to move away exploits the default AI of animals
+                                     *the set Velocity make sure that no knockback is given
+                                     */
+                                    if (firstRun) {
+                                        firstRun = false;
+                                        animal.damage(0, animal);
+                                        animal.setVelocity(new Vector());
+                                    } else {
+                                        animal.damage(0.5);
+                                    }
+                                }
+                            }
+                        }.runTaskTimer(this.plugin, 20, 20);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * OnPlayerInteract for Animal Overcrowding Control
+     *
+     * display a message about Animal Overcrowding Control
+     */
+    @EventHandler
+    public void onPlayerInteract(PlayerInteractEntityEvent event) {
+        Player player = event.getPlayer();
+        LivingEntity animal = (LivingEntity) event.getRightClicked();
+        World world = player.getWorld();
+
+        final boolean animalOverCrowdControl = CFG.getBoolean(RootNode.ANIMAL_OVERCROWD_CONTROL, world.getName());
+
+        if (animalOverCrowdControl && animal instanceof Animals 
+                && animal.getType() != EntityType.HORSE
+                && animal.getType() != EntityType.WOLF
+                && animal.getType() != EntityType.OCELOT) {
+
+            messenger.send(player, MessageNode.ANIMAL_OVERCROWD_CONTROL);
+        }
+    }
+
+    /**
+     * On Animal Death for Animal Overcrowding Control
+     *
+     * remove drops and exp from death cause not by player

```

Comment:
I believe there's a separate setting to not give exp from animals, regardless of death?

Also, if animals aren't killed by a player, I'm guessing exp is not dropped regardless.